packages/cdktf/lib/complex-computed-list.ts

```diff
@@ -136,29 +146,62 @@ export class ComplexComputedList extends ComplexComputedAttribute {
     }
 
     return this.terraformResource.interpolationForAttribute(
-      `${this.terraformAttribute}.${this.complexComputedListIndex}.${property}`
+      `${this.terraformAttribute}[${this.complexComputedListIndex}].${property}`
+    );
+  }
+}
+
+export abstract class ComplexList implements ITerraformAddressable {
+  constructor(
+    protected terraformResource: IInterpolatingParent,
+    protected terraformAttribute: string,
+    protected wrapsSet: boolean
+  ) {}
+
+  get fqn(): string {
+    return Token.asString(
+      this.terraformResource.interpolationForAttribute(this.terraformAttribute)
     );
   }
 }
 
 export class ComplexObject extends ComplexComputedAttribute {
+  /**
+   * @param terraformResource
+   * @param terraformAttribute
+   * @param complexObjectIndex the index of the complex object in a list
+   * @param complexObjectIsFromSet set to true if this item is from inside a set and needs tolist() for accessing it
+   *                               set to "0" for single list items
+   */
   constructor(
     protected terraformResource: IInterpolatingParent,
     protected terraformAttribute: string,
-    protected isSingleItem: boolean
+    protected complexObjectIndex: number,
+    protected complexObjectIsFromSet: boolean
   ) {
     super(terraformResource, terraformAttribute);
   }
 
   public interpolationForAttribute(property: string) {
+    if (this.complexObjectIsFromSet) {
+      return propertyAccess(
+        Fn.tolist(
+          this.terraformResource.interpolationForAttribute(
+            this.terraformAttribute
+          )
+        ),
+        [this.complexObjectIndex, property]
+      );
+    }
+
     return this.terraformResource.interpolationForAttribute(
-      `${this.terraformAttribute}${this.isSingleItem ? "[0]" : ""}.${property}`
+      `${this.terraformAttribute}[${this.complexObjectIndex}].${property}`
     );
   }
 
   protected interpolationAsList() {
     return this.terraformResource.interpolationForAttribute(
-      `${this.terraformAttribute}.*`
```

Comment:
Should this stick around or otherwise be exposed? I'm not sure where it's used currently.

Reply:
It is used here (but just to check the type, not actually called): https://github.com/hashicorp/terraform-cdk/blob/b0d5a7e5871bc4a5ca9674137482f1e79596704c/packages/cdktf/lib/tokens/private/encoding.ts#L162
But yeah, maybe I should just leave it as is, even if we thing it might not be used any longer.

Reply:
Reverted it.